src/daemon/daemonaccesscontrol.cpp

```diff
@@ -0,0 +1,137 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "daemonaccesscontrol.h"
+
+#if defined(MZ_MACOS)
+#  include <sys/types.h>
+#  include <unistd.h>
+#endif
+
+#include <QLocalSocket>
+
+#include "leakdetector.h"
+#include "logger.h"
+
+namespace {
+Logger logger("DaemonAccessControl");
+}  // namespace
+
+DaemonAccessControl::DaemonAccessControl() {
+  MZ_COUNT_CTOR(DaemonAccessControl);
+}
+
+DaemonAccessControl::~DaemonAccessControl() {
+  MZ_COUNT_DTOR(DaemonAccessControl);
+}
+
+bool DaemonAccessControl::isSessionActive() const {
+  return m_sessionOwner != -1;
+}
+
+bool DaemonAccessControl::startSession(QLocalSocket* socket) {
+  if (isSessionActive()) {
+    return false;
+  }
+
+  int peerId = getPeerId(socket);
+  if (peerId == -1) {
+    return false;
+  }
+
+  logger.debug() << "Session started by user" << peerId;
+  m_sessionOwner = peerId;
+  return true;
+}
+
+void DaemonAccessControl::resetSession() {
+  if (!isSessionActive()) {
+    return;
+  }
+
+  logger.debug() << "Session reset";
+  m_sessionOwner = -1;
+}
+
+bool DaemonAccessControl::authorizeCommandForPeer(const QString& command,
+                                                  QLocalSocket* socket) {
+  if (!isSessionActive()) {
+    if (command == "activate") {
+      if (!startSession(socket)) {
+        logger.error() << "Unable to start new session."
+                          "Can't authorize `activate` command.";
+
+        return false;
+      }
+
+      return true;
+    } else if (command == "status") {
+      // Other than "activate",
+      // only "status" command is available when there is no ongoing session.
```

Comment:
Could the comment be expanded to explain why "status" is allow while inactive? i.e. the client app asks for status when the vpn is not activated.